shared-bindings/_eve/__init__.c

```diff
@@ -45,94 +47,57 @@
 
 typedef struct _mp_obj__EVE_t {
     mp_obj_base_t base;
-    mp_obj_t dest[3];           // Own 'write' method, plus argument
-    int vscale;                 // fixed-point scaling used for Vertex2f
-    size_t n;                   // Current size of command buffer
-    uint8_t buf[512];           // Command buffer
+    common_hal__eve_t _eve;
 } mp_obj__EVE_t;
 
 STATIC const mp_obj_type_t _EVE_type;
 
-STATIC void _write(mp_obj__EVE_t *_EVE, mp_obj_t b) {
-    _EVE->dest[2] = b;
-    mp_call_method_n_kw(1, 0, _EVE->dest);
-}
+#define EVEHAL(s) \
+  (&((mp_obj__EVE_t*)mp_instance_cast_to_native_base((s), &_EVE_type))->_eve)
 
 STATIC mp_obj_t _register(mp_obj_t self, mp_obj_t o) {
```

Comment:
In order for these functions to [appear in the docs](https://circuitpython.readthedocs.io/en/latest/shared-bindings/_bleio/Characteristic.html) they'll need comments here similar to [these](https://github.com/adafruit/circuitpython/blob/master/shared-bindings/_bleio/Characteristic.c#L270). Although it is more work, I think it is worth it for the clarity.

Reply:
Yes, thanks, putting them together now.